src/or/rendservice.c

```diff
@@ -3596,7 +3596,7 @@ directory_post_to_hs_dir(rend_service_descriptor_t *renddesc,
         /* Don't upload descriptor if we succeeded in doing so last time. */
         continue;
       node = node_get_by_id(hs_dir->identity_digest);
-      if (!node || !node_has_descriptor(node)) {
+      if (!node || !node_has_preferred_descriptor(node,0)) {
```

Comment:
Does this need a comment about Tor2web, similar to the previous comments?

Reply:
I'm content to leave rend* alone as legacy code.